gobblin-runtime/src/main/java/org/apache/gobblin/util/MySQLStoreUtils.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.util;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+
+
+/**
+ * MySQL based implementations of stores require common functionality that can be stored in a utility class. The
+ * functionality includes executing prepared statements on a data source object and executing SQL queries at fixed
+ * intervals. The instantiater of the class should provide the data source used within this utility.
+ */
+public class MySQLStoreUtils {
+  private final DataSource dataSource;
+  private final Logger log;
+
+  public MySQLStoreUtils(DataSource dataSource, Logger log) {
+    this.dataSource = dataSource;
+    this.log = log;
+  }
+
+  /** `j.u.Function` variant for an operation that may @throw IOException or SQLException: preserves method signature checked exceptions */
+  @FunctionalInterface
+  public interface CheckedFunction<T, R> {
+    R apply(T t) throws IOException, SQLException;
+  }
+
+  /** Abstracts recurring pattern around resource management and exception re-mapping. */
+  public <T> T withPreparedStatement(String sql, CheckedFunction<PreparedStatement, T> f, boolean shouldCommit)
```

Comment:
what's your thinking on other classes with their own `withPreparedStatement`, such as `MysqlBaseSpecStore`--do they deserve a TODO comment because you recommend to migrate them to use this impl-in-common?

Reply:
Added the TODO comment there to refactor it if working on the class in the future.